go-controller/pkg/ovn/acl/acl.go

```diff
@@ -71,83 +32,41 @@ func RemoveACLFromNodeSwitches(switches []string, aclUUID string) error {
 	return nil
 }
 
-// RemoveACLFromPortGroup removes the ACL from the port-group
-func RemoveACLFromPortGroup(aclUUID, clusterPortGroupUUID string) error {
-	_, stderr, err := util.RunOVNNbctl("--", "--if-exists", "remove", "port_group", clusterPortGroupUUID, "acls", aclUUID)
-	if err != nil {
-		return errors.Wrapf(err, "Failed to remove reject ACL %s from port group %s: stderr: %q", aclUUID, clusterPortGroupUUID, stderr)
-	}
-	klog.Infof("ACL: %s, removed from the port group : %s", aclUUID, clusterPortGroupUUID)
-	return nil
-}
-
-// AddRejectACLToPortGroup adds a reject ACL to a PortGroup
-func AddRejectACLToPortGroup(clusterPortGroupUUID, aclName, sourceIP string, sourcePort int, proto v1.Protocol) (string, error) {
-	l3Prefix := "ip4"
-	if utilnet.IsIPv6String(sourceIP) {
-		l3Prefix = "ip6"
-	}
-
-	aclMatch := fmt.Sprintf("match=\"%s.dst==%s && %s && %s.dst==%d\"", l3Prefix, sourceIP,
-		strings.ToLower(string(proto)), strings.ToLower(string(proto)), sourcePort)
-	cmd := []string{"--id=@reject-acl", "create", "acl", "direction=" + types.DirectionFromLPort, "priority=" + types.DefaultDenyPriority, aclMatch, "action=reject",
-		fmt.Sprintf("name=%s", aclName), "--", "add", "port_group", clusterPortGroupUUID, "acls", "@reject-acl"}
-	aclUUID, stderr, err := util.RunOVNNbctl(cmd...)
-	if err != nil {
-		return "", errors.Wrapf(err, "Failed to add ACL: %s, %q, to cluster port group %s, stderr: %q", aclUUID, aclName, clusterPortGroupUUID, stderr)
-	}
-	return aclUUID, nil
-}
-
-// AddRejectACLToLogicalSwitch adds a reject ACL to a logical switch
-func AddRejectACLToLogicalSwitch(logicalSwitch, aclName, sourceIP string, sourcePort int, proto v1.Protocol) (string, error) {
-	l3Prefix := "ip4"
-	if utilnet.IsIPv6String(sourceIP) {
-		l3Prefix = "ip6"
-	}
-
-	aclMatch := fmt.Sprintf("match=\"%s.dst==%s && %s && %s.dst==%d\"", l3Prefix, sourceIP,
-		strings.ToLower(string(proto)), strings.ToLower(string(proto)), sourcePort)
-	cmd := []string{"--id=@reject-acl", "create", "acl", "direction=" + types.DirectionFromLPort, "priority=" + types.DefaultDenyPriority, aclMatch, "action=reject",
-		fmt.Sprintf("name=%s", aclName), "--", "add", "logical_switch", logicalSwitch, "acls", "@reject-acl"}
-
-	aclUUID, stderr, err := util.RunOVNNbctl(cmd...)
-	if err != nil {
-		return "", errors.Wrapf(err, "Failed to add ACL: %s, %q, to cluster switch %s, stderr: %q", aclUUID, aclName, logicalSwitch, stderr)
+func PurgeRejectRules(nbClient libovsdbclient.Client) error {
+	// TODO
+	// - Removing blindly reject ACLs might be problematic if in the future we add this types of ACLs for other purposes
+	// - Do ACLs need to be removed or only references to them, which is what we seem to be doing elsewhere?
```

Comment:
IIRC, deleting an object in OVSDB that is referenced somewhere else will result in an error.
You don't need to explicitly delete the ACL object, as once it's been removed from a Port Group, and there are no other references to it, ovsdb-server will automatically delete it.

Reply:
@dave-tucker Ok understood, I will remove the TODO. But question: does libovsdb test server automatically remove this unreferenced objects as well? I think it may not?